R/dependencies.R

```diff
@@ -87,17 +87,17 @@ snapshotRDependencies <- function(appDir, implicit_dependencies=c(), verbose = F
     default = c("duplicates")
   )
 
-  # get Bioconductor repos if any
-  biocRepos <- repos[grep('BioC', names(repos), perl = TRUE, value = TRUE)]
-  biocPackages <- if (length(biocRepos) > 0) {
-    available.packages(
-      contriburl = contrib.url(biocRepos, type = "source"),
-      type = "source",
-      filters = filters
-    )
-  }
+  # get Bioconductor repos if aniy
+  # We detect the presence of Bioconductor reopos by 
+  #   * existence of BiocManager amongst the installed packages
+  #   * the non-existence of any bioconductor repo in the CRAN like 
+  #            repo definitions 
+  if ("BiocManager" %in% installed.packages() && 
+    paste0("x",grep('BioC', names(options("repos")), perl = TRUE, value = TRUE)) == "x" ) {
+    
+    repos <- BiocManager::repositories() 
```

Comment:
Does this preserve the user's existing repositories if they've already customized them? What if you already have something like:

```
options(repos = c(RSPM = "https://rspm.company.org/repos/cran"))
```

Is that repository preserved?

Reply:
Yes, it preserves any other repo they may have defined, see: 

```
options()$repos; options(repos = c(RSPM = "https://rspm.company.org/repos/cran", RSPM2 = "https://rspm2.company.org/repos/cran")); paste("BiocManager"); BiocManager::repositories()
                                  RSPM                                  RSPM2 
 "https://rspm.company.org/repos/cran" "https://rspm2.company.org/repos/cran" 
[1] "BiocManager"
                                                BioCsoft                                                  BioCann 
           "https://bioconductor.org/packages/3.15/bioc" "https://bioconductor.org/packages/3.15/data/annotation" 
                                                 BioCexp                                            BioCworkflows 
"https://bioconductor.org/packages/3.15/data/experiment"       "https://bioconductor.org/packages/3.15/workflows" 
                                               BioCbooks                                                     RSPM 
          "https://bioconductor.org/packages/3.15/books"                    "https://rspm.company.org/repos/cran" 
                                                   RSPM2 
                  "https://rspm2.company.org/repos/cran" 
```

Reply:
Great, thanks for confirming!